### PR TITLE
Protect payment filters AJAX metadata endpoint

### PR DESCRIPTION
**Pull request by dhrupo**
## What does this PR do and why?

This PR fixes an authenticated metadata disclosure in the AJAX endpoint `fluentform_get_all_payments_entries_filters`.

Before this change, `PaymentEntries::getFilters()` returned payment filter metadata for any logged-in WordPress user because it had no Fluent Forms capability guard and no nonce/capability verification flow. A low-privileged user could call the endpoint directly and enumerate payment workflow metadata.

## Security impact

Before fix:
- Any authenticated user could query available payment transaction types, statuses, payment methods, and form titles/IDs with payment activity.
- This leaked payment-related business metadata to users who do not have payment-view permissions.

After fix:
- `getFilters()` now enforces `Acl::verify('fluentform_view_payments', form_id)`.
- This applies the same permission and nonce verification model already used by `getPayments()`.
- Unauthorized callers are blocked.

## Root cause

`getFilters()` had no authorization gate, while `getPayments()` correctly required `fluentform_view_payments`.

## What changed

File changed:
- `app/Modules/Payments/Classes/PaymentEntries.php`

Changes made in `getFilters()`:
1. Read and sanitize request attributes, including `form_id` via `Acl::normalizeFormId`.
2. Add `Acl::verify('fluentform_view_payments', form_id)` to enforce capability + nonce verification flow.
3. Apply consistent query scoping for all grouped metadata queries (`transaction_type`, `status`, `forms`, `payment_method`):
   - filter by selected `form_id` when provided.
   - filter by `fluentform/current_user_allowed_forms` when available.

## How to reproduce (before fix)

1. Log in with a low-privileged WordPress user (for example `subscriber`) without Fluent Forms payment permissions.
2. Send an authenticated request:

```http
POST /wp-admin/admin-ajax.php
Content-Type: application/x-www-form-urlencoded

action=fluentform_get_all_payments_entries_filters
```

3. Observe JSON response includes metadata such as:
- `available_payment_types`
- `available_statuses`
- `available_forms`
- `available_methods`

## How to verify (after fix)

1. Repeat the same request as a low-privileged user.
2. Confirm the request is rejected by ACL verification.
3. As an authorized user with `fluentform_view_payments`, open Payment Entries UI and verify filters still load correctly.
4. Optionally pass a specific `form_id` and verify all filter groups remain scoped to that form/allowed forms.

## Regression notes

- No frontend changes required.
- Existing authorized Payment Entries behavior remains intact.
- Fix tightens only unauthorized access and inconsistent metadata scoping.

## Checks run

- `php -l app/Modules/Payments/Classes/PaymentEntries.php`
- `git diff --check dev..HEAD`

### CODE REVIEW

**Review by autharif — APPROVED** [review]
<h3>Summary</h3>
<p>The change looks ready to merge with no blocker-level findings.</p>
<p><strong>Key changes:</strong></p>
<ul>
<li><code>getFilters()</code> now normalizes incoming <code>form_id</code> via <code>Acl::normalizeFormId</code> through <code>fluentform_backend_sanitizer</code>, which keeps request-to-query flow aligned with existing backend sanitization patterns (<code>app/Modules/Payments/Classes/PaymentEntries.php</code>).</li>
<li>Authorization is performed inside <code>getFilters()</code> with <code>Acl::verify(&#39;fluentform_view_payments&#39;, ArrayHelper::get($attributes, &#39;form_id&#39;))</code>, creating an explicit access-control checkpoint before building filter datasets (<code>app/Modules/Payments/Classes/PaymentEntries.php</code>).</li>
</ul>
<h3>Confidence Score: 4/5</h3>
<ul>
<li>Merge stance: <code>APPROVE</code> with no confirmed Critical or Important issues.</li>
<li>The updated filter path shows a clear and consistent chain from request attributes to normalization, authorization, and scoped query building in one place. The implementation also preserves the existing allowed-forms filter contract for role-based form scoping.</li>
</ul>
<p><sub>Last reviewed commit: <a href="https://github.com/fluentform/fluentform/commit/74038d50be6c58d77b91135ecfc93e70d93483af">Fix: Protect payment filters AJAX metadata endpoint</a></sub></p>
